lib/node_modules/@stdlib/math/base/special/factorial2/lib/factorial2.js

```diff
@@ -0,0 +1,85 @@
+'use strict';
+
+// MODULES //
+
+var exp2 = require( '@stdlib/math/base/special/exp2' );
+var gamma = require( '@stdlib/math/base/special/gamma' );
+var isnan = require( '@stdlib/math/base/utils/is-nan' );
+var isInteger = require( '@stdlib/math/base/utils/is-integer' );
+var isNegativeZero = require( '@stdlib/math/base/utils/is-negative-zero' );
+var sqrt = require( '@stdlib/math/base/special/sqrt' );
+var DOUBLE_FACTORIALS = require( './doublefactorials.json' );
+var NINF = require( '@stdlib/math/constants/float64-ninf' );
+var PINF = require( '@stdlib/math/constants/float64-pinf' );
+
+// VARIABLES //
+
+var MAX_FACTORIAL = 170; // TODO: consider extracting as a constant
+var PIO2 = 1.57079632679489661923; // pi/2
+
+
+// MAIN //
+
+/**
+* Evaluates the double factorial of `x`.
+*
+* @param {number} x - input value
+* @returns {number} double factorial
+*
+* @example
+* var v = factorial2( 3.0 );
+* // returns 3.0
+* @example
+* var v = factorial2( -1.5 );
+* // returns ~-1.381
+* @example
+* var v = factorial2( -0.5 );
+* // returns ~-0.963
+* @example
+* var v = factorial2( 0.5 );
+* // returns ~0.963
+* @example
+* var v = factorial2( -10.0 );
+* // returns NaN
+* @example
+* var v = factorial2( 171.0 );
+* // returns Number.POSITIVE_INFINITY
+* @example
+* var v = factorial2( NaN );
+* // returns NaN
+*/
+function factorial2( x ) {
+	var n;
+	if (
+		(isInteger( x ) && x < 0) ||
```

Comment:
Alternatively, keep `isInteger`, but cache the result so you can use the `boolean` later at line `66`.

``` javascript
var isInt;

isInt = isInteger( x );
if ( isInt && x < 0 ) {
   return NaN;
}
...
if ( isInt ) {
  ...
}
```